### DISCUSSION TITLE
Provide parameter for set_node_attributes() to throw key errors

### DISCUSSION THREAD

**Discussion by AlexHodgson** [networkx/networkx#7296] [Ideas]
Hello all,

I see there has been discussion on this topic in the past (https://github.com/networkx/networkx/issues/2434, https://github.com/networkx/networkx/issues/4341), and the decision was made to fail silently if you attempt to set an attribute on a node that doesn't exist in the graph. This seems rather risky and could lead to some difficult to find bugs if someone was expcting that their values are getting mapped, and was unaware of the unusual error handling of this function.
For my current usecase it is essential every attribute I pass in gets assigned to a node, and I would get errornous results if a node is missing the attribute. So I would like to be sure that everything I pass in has been assigned, and en error to be thrown if I made a mistake in labelling the attributes. I see there were valid reasons for choosing to hide this error, but would it be possible to add an optional paramater to raise the exception if `raise_errors=True` for example.

I had a look at the function and this would be a super quick change so I can make a PR if the community agrees.

**Comment by rossbar** [comment]
Since dict keys support some set-like operations, checking that all attributes in the key dictionary are present in the Graph should be relatively straightforward, something like:

```python
>>> G = nx.Graph()
>>> G.add_node(0)
>>> attr_dict = {0: {'name': 'foo'}, 1: {'name': 'bar'}}
>>> len(attr_dict.keys() - set(G))  # should == 0 for the case you're describing
```

Since this is a one-liner, my preference would be that performing this check is on the user. OTOH, since the user has to *know* that `set_node_attributes` silently ignores keys that are not in `G`, I can see an argument for expanding the signature. Is your use-case covered by the above, or did you have something else in mind?
